--- a/nss_dp_main.c
+++ b/nss_dp_main.c
@@ -403,6 +403,8 @@ static int nss_dp_close(struct net_devic
 	return 0;
 }
 
+static void nss_dp_adjust_link(struct net_device *netdev);
+
 /*
  * nss_dp_open()
  */
@@ -502,6 +504,7 @@ static int nss_dp_open(struct net_device
 		dp_priv->link_state = __NSS_DP_LINK_DOWN;
 		phy_start(dp_priv->phydev);
 		phy_start_aneg(dp_priv->phydev);
+		nss_dp_adjust_link(netdev);
 	}
 
 	return 0;
